apis/v1beta1/validation/httproute_test.go

```diff
@@ -982,3 +982,137 @@ func TestValidateHTTPRouteTypeMatchesField(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateHTTPParentRefs(t *testing.T) {
```

Comment:
can you please add a couple more test cases like 
* different listeners having the same `sectionName`. (it should pass)
* the same listener referenced twice; one reference does have the sectionName, but the other doesn't. (it shouldn't pass)

Reply:
> the same listener referenced twice; one reference does have the sectionName, but the other doesn't. (it shouldn't pass)

In my current implementation, this case will pass because it assumes their sectionNames are different. 
But I think this case is correct, I will add some checks to make sure it happens.

Reply:
I add more checks when the same listener referenced twice:
```
		if s, ok := parentRefsSectionMap[targetParentRefs]; ok {
			if len(s[0]) == 0 || len(*section) == 0 {
				errs = append(errs, field.Required(path, "ParentRefs section names must all be set when ParentRefs includes 2 or more references to the same parent"))
				return errs
			}
			if utils.ContainsInSectionNameSlice(s, section) {
				errs = append(errs, field.Invalid(path.Index(i).Child("parentRefs").Child("sectionName"), section, "must be unique when ParentRefs includes 2 or more references to the same parent"))
				return errs
			}
		}
```
It means if the first ParentRefs section name is not set, or current ParentRefs section name is not set, it will also raise an error.